docs/typescript/index.md

```diff
@@ -0,0 +1,90 @@
+---
+order: 3
+---
+
+# Using with TypeScript / Glint
+
+This addon is written in TypeScript, and strives to give TypeScript users a first-class experience. It will be mostly used within templates, which TypeScript does not understand unfortunately. Here [Glint](https://github.com/typed-ember/glint) comes to the rescue, which will make strict type checking available in templates.
+
+> While this library adheres and strives for [SemVer](https://semver.org/), TypeScript does not, and Glint still has no stable release. So this addon will not declare its TypeScript/Glint support to be stable _yet_.
+> Type changes during minor upgrades may be considered breaking changes to some.
+> The goal, is to follow the advice of [Semantic Versioning for TypeScript Types](https://www.semver-ts.org/) eventually. Until noted in the Changelog that our TypeScript types fall under SemVer, consider type changes as "bugfixes" for patch releases.
+
+All the components, which are `<HeadlessForm>` as well as those [yielded from it](./usage), have strict Glint types. In case you are already using _strict mode templates_ via [ember-template-imports](https://github.com/ember-template-imports/ember-template-imports), you only need to import the component and you are ready to go. For the more likely case of using classic `.hbs` templates, Glint requires a template registry to be set up, that declares a mapping of component _names_ to their types. This addon follows the [recommendations for Glint-enabled addons](https://typed-ember.gitbook.io/glint/using-glint/ember/using-addons#using-glint-enabled-addons), so add the addon's provided registry to your app's own registry as follows:
```

Comment:
Suggestion: I think there should be a header (h2?) above line 13 with something like, "Usage with classic templates (hbs)"

Reason being, it will flag it for folks who are using `hbs` that they need to do this in order to "see" the HeadlessForm types